packages/driver/cypress/integration/e2e/multi-domain/basic_login_spec.ts

```diff
@@ -48,132 +34,250 @@ describe('basic login', { experimentalSessionSupport: true, experimentalMultiDom
       .invoke('text')
       .should('equal', 'Welcome FJohnson')
     })
+
+    // TODO: this test 'passes' but acts strangely by showing no log and fails the subsequent test
+    it.skip('visits foobar first', () => {
+      cy.visit('http://www.foobar.com:3500/fixtures/auth/index.html') // Establishes Primary Domain
+      cy.get('[data-cy="login"]').click() // Takes you to identityProvider.com
+      cy.switchToDomain('idp.com', () => {
+        cy.get('[data-cy="username"]').type('BJohnson')
+        cy.get('[data-cy="login"]').click()
+      }) // Trailing edge wait, waiting to return to the primary domain
+
+      // Verify that the user has logged in on /siteA
+      cy.get('[data-cy="welcome"]')
+      .invoke('text')
+      .should('equal', 'Welcome BJohnson')
+    })
   })
 
   // Scenario, Token based auth. Visit IDP hosted on secondary domain, login and redirect back to site.
+  // @ts-ignore
   describe('visit secondary first', () => {
-    // Problem: where does the primary domain get established
-    it('logs in with idp redirect', () => {
-      cy.window().then((win) => {
-        win.location.href = 'http://www.foobar.com:3500/fixtures/auth/idp.html'
+    describe('How to determine primary', () => {
+      // Enable to set the top domain to example.com before running the next test.
+      // it('example.com', () => {
+      //   cy.visit('http://www.example.com')
+      // })
+
+      // Problem: where does the primary domain get established
+      // @ts-ignore
+      // TODO: This test will fail
```

Comment:
Does it make sense to combine the two comments. Something like this?:
```suggestion
      // TODO: This test will fail. Where does the primary domain get established?
```

Reply:
updated